pkg/downloader/chart_downloader.go

```diff
@@ -392,3 +466,7 @@ func loadRepoConfig(file string) (*repo.File, error) {
 	}
 	return r, nil
 }
+
+func GetInMemoryCache() InMemoryChartList {
```

Comment:
This still exposes the details of the cache.  All the client needs is a `*chart.Chart` to be returned.
Please replace this method with:
```
func GetChart(chart string) *chart.Chart, error {
     chartData := inMemoryCache[chart].ChartData
     return loader.LoadArchive(chartData)
}
```

But I would still like this to be part of the `loader` package somehow, if possible.

Reply:
I have addred the function to Get the Chart from the Cache only. I tried moving this function to loader. Considering that the downloader downloads and stores the Chart Data locally in the package, we would have to use a GetChart in Loader which gets it from the Downloader and serves it to the Client Side. The issue happening here is that it is creating an import cycle and helm won't build after.